### ISSUE TITLE
findOneAndUpdate don't cast _bsontype == "Long" to Number

### ISSUE THREAD

**Issue by truestep** [mongoosejs/mongoose-long#10]
using:

```
    "mongodb": "^3.5.2",
    "mongoose": "^5.8.11",
    "mongoose-long": "^0.2.1",
```

```
  async update(doc) {
    let obj = (await this.do.findOneAndUpdate({_id: doc._id}, doc, {new: true}).exec())
    obj = obj.toObject(obj)
    return obj
}
```
 instead of Number inside the retreived doc you'll get

```
 { 
  _bsontype: "Long",
  low:_ 1971555403
  high_: 369
 }

```
So I need to go recursively and convert to Number myself

```
_toObject(originalObj) {
    const getProps = function (obj) {
      for (let property in obj) {
        if (obj.hasOwnProperty(property) && obj[property] != null) {
          if (obj[property].constructor == Object) {
            getProps(obj[property])
          } else if (obj[property].constructor == Array) {
            for (let i = 0; i < obj[property].length; i++) {
              getProps(obj[property][i])
            }
          } else {
            if (obj[property]._bsontype === "Long") {
              obj[property] = Number((obj[property]))
              console.log(obj[property])
            }
          }
        }
      }
    }
    originalObj = originalObj.toObject()
    getProps(originalObj)
    return originalObj
  }
```
findById works fine

**Comment by vkarpov15** [comment]
This is expected behavior: Mongoose's `toObject()` doesn't convert MongoDB-specific objects to primitives. For example, `toObject()` leaves ObjectIds as ObjectIds rather than strings. Same for longs.

However, there's a much neater fix for this with Mongoose 5.9 and mongoose-long@0.3.0 using default options per schema type: https://thecodebarbarian.com/whats-new-in-mongoose-59-schematype-default-options-and-per-document-populate.html#default-options-per-schematype

```javascript
'use strict';
  
const mongoose = require('mongoose');

require('mongoose-long')(mongoose);

const { Schema } = mongoose;

run().catch(err => console.log(err));

async function run() {
  await mongoose.connect('mongodb://localhost:27017/test', {
    useNewUrlParser: true,
    useUnifiedTopology: true
  });

  await mongoose.connection.dropDatabase();

  // Add a default transform to all longs that transforms them into strings.
  // Transforms are only applied when you call `toObject()` or `toJSON()`
  mongoose.Schema.Types.Long.set('transform', v => v.toString());

  const schema = Schema({ myLong: 'Long' });
  const Model = mongoose.model('Test', schema);

  await Model.create({ myLong: 42 });
  const doc = await Model.findOne();
  console.log(doc.toObject().myLong); // 42 as a string
}
```

**Comment by apos-cadabra** [comment]
I've tried your example locally and getting

**Long { _bsontype: 'Long', low_: 42, high_: 0 }**

as an output in console.log
using Mongoose 5.9.23 and mongoose-long@0.3.0

**Comment by vkarpov15** [comment]
@apos-cadabra you're right, that's my mistake. Fixed in v0.3.1 :+1:
